ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractArcticOptimizePlan.java

```diff
@@ -174,7 +178,129 @@ protected BasicOptimizeTask buildOptimizeTask(@Nullable List<DataTreeNode> sourc
     return optimizeTask;
   }
 
-  public boolean tableNeedPlan() {
+  @Override
+  protected void addOptimizeFiles() {
+    addChangeFilesIntoFileTree();
+    addBaseFilesIntoFileTree();
+    completeTree();
+  }
+
+  private void completeTree() {
+    partitionFileTree.values().forEach(FileTree::completeTree);
+  }
+
+  protected void addChangeFilesIntoFileTree() {
+    
+  }
+
+  protected void addBaseFilesIntoFileTree() {
+    LOG.debug("{} start {} plan base files", tableId(), getOptimizeType());
+    UnkeyedTable baseTable = getBaseTable();
+
+    Set<String> baseFileSet = new HashSet<>();
+    Set<String> deleteFileSet = new HashSet<>();
+    baseFileScanTasks.forEach(task -> {
+      DataFile baseFile = task.file();
+      List<DeleteFile> deletes = task.deletes();
+      String partition = baseTable.spec().partitionToPath(baseFile.partition());
+
+      currentPartitions.add(partition);
+      // TODO remove this
```

Comment:
To remove 
```
      if (anyTaskRunning(partition)) {
        return;
      }
...

  protected boolean anyTaskRunning(String partition) {
    return partitionTaskRunning.get(partition) != null && partitionTaskRunning.get(partition);
  }
```

It's redundant to check any tasks running for each partition during the plan, since we have checked it before the plan.